pennylane/transforms/qcut.py

```diff
@@ -284,3 +286,77 @@ def fragment_graph(graph: MultiDiGraph) -> Tuple[Tuple[MultiDiGraph], MultiDiGra
         communication_graph.add_edge(start_fragment, end_fragment, pair=(node1, node2, wire))
 
     return subgraphs, communication_graph
+
+
+def _find_new_wire(wires: Wires) -> int:
+    """Finds a new wire label that is not in ``wires``."""
+    ctr = 0
+    while ctr in wires:
+        ctr += 1
+    return ctr
+
+
+def graph_to_tape(graph: MultiDiGraph) -> QuantumTape:
+    """
+    Converts a directed multigraph to the corresponding quantum tape.
+
```

Comment:
Perhaps mention that each node in `graph` should have an `order` attribute specifying the topological order of the operations.

Although this sounds a bit weird given that a digraph should already specify the order. Instead, maybe we just add a note similar to [here](https://github.com/PennyLaneAI/pennylane/pull/2158/files#diff-80b23fd1a16334ed3be95fed079f293972ee780704fc083dacc652d3f4c374ddR240) showing that this function is part of a larger pipeline.

Reply:
Though it might be nice to mention that we accommodate for mid-circuit measurements (when specified as a `MeasureNode`)